charts/external-dns/templates/deployment.yaml

```diff
@@ -132,9 +137,51 @@ spec:
           resources:
             {{- toYaml . | nindent 12 }}
           {{- end }}
-      {{- if or .Values.secretConfiguration.enabled .Values.extraVolumes }}
+        {{- if contains "/" $providername }}
+        - name: provider
+          {{- with .Values.securityContext }}
+          securityContext:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          image: {{ tpl $providername $ }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- with .Values.provider.env }}
+          env:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          args:
+          {{- range .Values.provider.args }}
+            - {{ tpl . $ }}
+          {{- end }}
+          ports:
+            - name: provider-metrics
```

Comment:
If there are metrics available from the provider, should there be an optional podmonitor for it as well?

Reply:
We should probably extend the existing servicemonitor to also hit provider-metrics when using a webhook provider.